src/solc_0.8/polygon/child/land/PolygonLandBaseToken.sol

```diff
@@ -0,0 +1,386 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.2;
+
+import "@openzeppelin/contracts-0.8/utils/Address.sol";
+import "../../../common/BaseWithStorage/ERC721BaseToken.sol";
+
+contract PolygonLandBaseToken is ERC721BaseToken {
+    using Address for address;
+
+    uint256 internal constant GRID_SIZE = 408;
+
+    uint256 internal constant LAYER = 0xFF00000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_1x1 = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_3x3 = 0x0100000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_6x6 = 0x0200000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_12x12 = 0x0300000000000000000000000000000000000000000000000000000000000000;
+    uint256 internal constant LAYER_24x24 = 0x0400000000000000000000000000000000000000000000000000000000000000;
+
+    /**
+     * @notice Return the name of the token contract
+     * @return The name of the token contract
+     */
+    function name() external pure returns (string memory) {
+        return "Sandbox's LANDs";
+    }
+
+    /**
+     * @notice Return the symbol of the token contract
+     * @return The symbol of the token contract
+     */
+    function symbol() external pure returns (string memory) {
+        return "LAND";
+    }
+
+    // solium-disable-next-line security/no-assign-params
+    function uint2str(uint256 _i) internal pure returns (string memory _uintAsString) {
+        if (_i == 0) {
+            return "0";
+        }
+        uint256 j = _i;
+        uint256 len;
+        while (j != 0) {
+            len++;
+            j /= 10;
+        }
+        bytes memory bstr = new bytes(len);
+        uint256 k = len;
+        while (_i != 0) {
+            k = k - 1;
+            uint8 temp = (48 + uint8(_i - (_i / 10) * 10));
+            bytes1 b1 = bytes1(temp);
+            bstr[k] = b1;
+            _i /= 10;
+        }
+        return string(bstr);
+    }
+
+    /**
+     * @notice Return the URI of a specific token
+     * @param id The id of the token
+     * @return The URI of the token
+     */
+    function tokenURI(uint256 id) public view returns (string memory) {
```

Comment:
Maybe you could do a unit test for this function since we modify the uint2str function